lsm-tree.cabal

```diff
@@ -94,22 +94,22 @@ library
     Database.LSMTree.Normal
 
   build-depends:
-    , base                  >=4.16      && <4.20
+    , base                  >=4.16      && <4.21
     , bitvec                ^>=1.1
     , bytestring            ^>=0.11.4.0 || ^>=0.12.1.0
     , containers
     , crc32c                ^>=0.2.1
     , deepseq               ^>=1.4      || ^>=1.5
     , filepath
     , fs-api                ^>=0.2
-    , io-classes            ^>=1.4
+    , io-classes            ^>=1.4      || ^>=1.5
```

Comment:
is there a reason to allow older `io-classes`? If there isn't, it's better to drop it as CI won't test it. Ditto for other dependencies.

Reply:
`io-classes` will build for all supported GHC versions if and only if we remove the conditional dependency relaxation from the `cabal.project` file:
```
-- if impl(ghc >=9.10)                                                                                          
```

Otherwise, the `fs-api` and `fs-sim` packages reject the newer version of `io-classes` specified in the package description.
```
$ cabal build --with-compiler=ghc-9.2.8
Resolving dependencies...
Error: [Cabal-7107]
Could not resolve dependencies:
[__0] trying: fs-api-0.2.0.1 (user goal)
[__1] next goal: io-classes (dependency of fs-api)
[__1] rejecting: io-classes-1.5.0.0 (conflict: fs-api => io-classes>=0.3 && <1.5)
[__1] trying: io-classes-1.4.1.0
[__2] next goal: lsm-tree (user goal)
[__2] rejecting: lsm-tree-0.1.0.0 (conflict: io-classes==1.4.1.0, lsm-tree => io-classes^>=1.5)
```

I can either leave the expanded version ranges in the package description, or relax the constraints in the `cabal.project` file until the upper bounds of `base` in `fs-api` and `fs-sim` get bumped.

Reply:
For the time being I made the`allow-newer` stanza, which was previous conditional on GHC 9.10, now apply universally to all GHC version with the intention of removing the stanza after a new version of `fs-api` and `fs-sim` are accessible.

Reply:
Never-mind, I didn't notice that the `fs-api` and `fs-sim` versions were explicitly pinned to a commit hash. I can update the commit hash to a newer version which supports GHC 9.10.